File: frontend/client/src/app/shared/models/product-color-variant.ts
import { ProductVariant } from './product-variant';
import { ProductVariantImage } from './product-variant-image';

export interface ProductColorVariant {
  id: string;
  productId: string;
  productSlug: string;
colorOptionId: string;
colorName: string;
  slug: string;
  price: number;
  discountPrice?: number;

  variants: ProductVariant[];
  images: ProductVariantImage[];
}

File: Slothsy.Domain/Entities/ProductColorVariant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Slothsy.Domain.Entities
{
    public class ProductColorVariant
    {
        public Guid Id { get; set; }

        public Guid ProductId { get; set; }
        public Product Product { get; set; } = null!;

        public Guid ColorOptionId { get; set; }
        public ColorOption ColorOption { get; set; } = null!;

        public string? Slug { get; set; }

        public decimal Price { get; set; }
        public decimal? DiscountPrice { get; set; }

        public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();

        public ICollection<ProductVariantImage> Images { get; set; } = new List<ProductVariantImage>();
    }
}
